File: src/main/java/net/loxal/soa/restkit/endpoint/Endpoint.java
/*
 * Copyright 2014 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint;

import javax.ws.rs.core.Response;
import java.text.MessageFormat;
import java.util.logging.Logger;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public abstract class Endpoint {
    public static final String URI_PATH_SEPARATOR = "/";
    protected static final int ASYNC_RESPONSE_TIMEOUT = 3;
    protected static final String ID_PATH_PARAM = "id";
    protected static final String ID_PATH_PARAM_PLACEHOLDER = "{" + ID_PATH_PARAM + "}";
    protected static final Logger LOG = Logger.getGlobal();
    private static final String ENTITY_ID_GROUP = "entityIdGroup";
    private static final Pattern ENTITY_ID_PATTERN = Pattern.compile(MessageFormat.format("/(?<{0}>[0-9a-f]+)$", ENTITY_ID_GROUP));

    protected String extractIdOfLocation(Response createdVote) {
        Matcher entityId = ENTITY_ID_PATTERN.matcher(createdVote.getLocation().getPath());

        final String id;
        if (entityId.find()) {
            id = entityId.group(ENTITY_ID_GROUP);
        } else {
            id = "";
        }
        return id;
    }
}

File: src/main/kotlin/net/loxal/soa/restkit/endpoint/Endpoint.kt
/*
 * Copyright 2017 Alexander Orlov <alexander.orlov@loxal.net>. All rights reserved.
 */

package net.loxal.soa.restkit.endpoint

import org.slf4j.Logger
import org.slf4j.LoggerFactory

abstract class Endpoint protected constructor() {
    companion object {
        val URI_PATH_SEPARATOR: String = "/"
        const val ASYNC_RESPONSE_TIMEOUT: Int = 3
        const val ID_PATH_PARAM: String = "id"
        const val ID_PATH_PARAM_PLACEHOLDER: String = "{$ID_PATH_PARAM}"
        val LOG: Logger = LoggerFactory.getLogger(Endpoint::class.java)
    }
}